### ISSUE TITLE
Add span name to tags when you use trace to metrics query

### ISSUE THREAD

**Issue by oscaratienzaatosnet** [grafana/tempo#1970]
I think it is necessary to add span name to tags when you use trace to metrics feature as apm metrics are based on span name, and is necessary to see that metrics.

so im expecting the posibility to add span name to  tags when they are expanded.

Using other things is a workaround that is very dificult from the point of just query.

Thank you

**Comment by mapno** [comment]
Hi @oscaratienzaatosnet, the span name is extracted from the intrinsic span property `name`. There is no need to add an additional tag for the span name. Does that answer you question?

**Comment by oscaratienzaatosnet** [comment]
it is supposed to do as this, but how can i send the span name to the metric query? can you test (please) yourself?

**Comment by mapno** [comment]
The span name is automatically added to the metrics as the label `span_name`. Not sure I understand you're question.

There are multiple [examples](https://github.com/grafana/tempo/tree/main/example) you can run to see how spanmetrics looks like.

**Comment by oscaratienzaatosnet** [comment]
i mean in this part


![image](https://user-images.githubusercontent.com/114416250/210526242-b2fd54ad-c23e-4456-9dac-6d6464e23384.png)


how can i add span_name to $__tags as it is not included by default

**Comment by mapno** [comment]
Ah, I understand now sorry. Unfortunately, it's not possible to add intrinsic attributes to 'Tags'. Only resource/span attributes.

One thing you could do is add the name as an attribute as well, with something like the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor). I believe this config will do it:
```
transform:
  trace_statements:
    - context: span
      statements:
        - set(attributes["span_name"], name)
```

**Comment by oscaratienzaatosnet** [comment]
i dont think it is a question of adding a new attribute to the span, as the span has the name per default. this is a workaround that is ok (although we are not using otel collector, and we are sending directly our traces to our tempo). 

That is why this is a feature request, to add the possibility of pass the span name to the query, as apm metrics are based in span_name, and getting this name from atributes requires a bit of calculation.

**Comment by joe-elliott** [comment]
This is a Grafana feature request. Moving this to the appropriate repo.

Heads up, @grafana/observability-traces-and-profiling.
